File: boj/1026.py
# 출처 : https://www.acmicpc.net/problem/1026
n = int(input())
a = list(map(int, input().split()))
b = list(map(int, input().split()))

# 가장 작은 값을 만들기 위해서는 가장 큰수를 가장 작은 수에 곱해야 하므로
# b는 내림 차순, a는 오름차순으로 정렬한다.ㄴ
a.sort()
b.sort(reverse=True)
answer = 0
for i in range(n):
    answer += a[i]*b[i]

print(answer)

File: js/1026.js
// 출처 : https://www.acmicpc.net/problem/1026
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const input = [];
rl.on("line", function (line) {
    input.push(line)
}).on("close", function () {
    solution(input);
    process.exit();
});

function solution(inputs) {
    const n = parseInt(inputs[0])
    let a = []
    let b = []
    inputs[1].split(' ').forEach(data => {
        a.push(parseInt(data));
    })
    inputs[2].split(' ').forEach(data => {
        b.push(parseInt(data));
    })


    // console.log(a);
    a = a.sort((a, b) => a - b);
    b = b.sort((a, b) => b - a);
    // console.log(a);
    // console.log(b);
    let answer = 0;
    for (let i = 0; i < n; i++) {
        answer += a[i] * b[i];
    }
    console.log(answer);
}